hollow/src/main/java/com/netflix/hollow/api/client/StaleHollowReferenceDetector.java

```diff
@@ -111,27 +113,51 @@ private synchronized void housekeeping() {
     }
 
     public void startMonitoring() {
-        Thread t = new Thread(new Runnable() {
-            public void run() {
-                while(true) {
-                    try {
-                        Thread.sleep(HOUSEKEEPING_INTERVAL);
-                    } catch (InterruptedException e) { }
-                    housekeeping();
-
-                    detector.staleReferenceExistenceDetected(countStaleReferenceExistenceSignals());
-                    detector.staleReferenceUsageDetected(countStaleReferenceUsageSignals());
-                }
-            }
-        });
-        t.setDaemon(true);
-        t.start();
+        if (monitor == null) {
+            monitor = new Thread(new Monitor(this));
+            monitor.setDaemon(true);
+            monitor.start();
+        }
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
```

Comment:
It's often preferable to avoid finalize methods if possible.  In this case the monitor thread could complete after the weak reference is cleared instead of relying on the interrupted status set by the finalize method. Then no additional field is required.